src/ethereum/eth.js

```diff
@@ -156,7 +141,6 @@ export const eth = {
    * @return {object}      - An object describing the transaction (if it exists)
    */
   fetchTxStatus(txId) {
```

Comment:
I'm not entirely sure where methods like this or `fetchTxReceipt` should live. Maybe it's best to forward everything down to the wallet and do:

`this.wallet.fetchTxReceipt()`

or, to avoid coupling the `Wallet` to `Contract`:

`Contract.transaction(this.wallet.fetchTxReceipt, txId)`

but it feels like an indirection. Furthermore, every time we need to add a method interfacing with `web3` we'll need to add it both to `eth` and to `Wallet`.

I'm torn about this and I don't know exactly what to do. Maybe the problem is the `Wallet` abstraction